core/dotnet/src/HttpCore.cs

```diff
@@ -47,6 +43,69 @@ public async Task<ModelType> SendAsync<ModelType>(RequestInfo requestInfo, IResp
             else
                 return await responseHandler.HandleResponseAsync<HttpResponseMessage, ModelType>(response);
         }
+        public async Task<ModelType> SendPrimitiveAsync<ModelType>(RequestInfo requestInfo, IResponseHandler responseHandler = default) {
+            if(requestInfo == null)
+                throw new ArgumentNullException(nameof(requestInfo));
+
+            await AddBearerIfNotPresent(requestInfo);
+            
+            using var message = GetRequestMessageFromRequestInfo(requestInfo);
+            var response = await this.client.SendAsync(message);
+            if(response == null)
+                throw new InvalidOperationException("Could not get a response after calling the service");
+            if(responseHandler == null) {
+                using var contentStream = await response.Content.ReadAsStreamAsync();
+                var modelType = typeof(ModelType);
+                if(modelType == typeof(Stream)) {
+                    return (ModelType)(contentStream as object);
+                } else {
+                    using var jsonDocument = JsonDocument.Parse(contentStream);
+                    var rootNode = new JsonParseNode(jsonDocument.RootElement);
+                    response.Dispose();
+                    requestInfo.Content?.Dispose();
+                    if(modelType == typeof(bool)) {
+                        return (ModelType)(rootNode.GetBoolValue() as object);
+                    } else if(modelType == typeof(string)) {
+                        return (ModelType)(rootNode.GetStringValue() as object);
+                    } else if(modelType == typeof(int)) {
+                        return (ModelType)(rootNode.GetIntValue() as object);
+                    } else if(modelType == typeof(float)) {
+                        return (ModelType)(rootNode.GetFloatValue() as object);
+                    } else if(modelType == typeof(double)) {
+                        return (ModelType)(rootNode.GetDoubleValue() as object);
+                    } else if(modelType == typeof(Guid)) {
+                        return (ModelType)(rootNode.GetGuidValue() as object);
+                    } else if(modelType == typeof(DateTimeOffset)) {
+                        return (ModelType)(rootNode.GetDateTimeOffsetValue() as object);
+                    } else throw new InvalidOperationException("error handling the response, unexpected type");
```

Comment:
consider extracting this to its own method and you can avoid "as + cast" with the temp variable.

I wish there was a clean way of mapping the type into the specific get method, but all the alternatives I could find are less readable then this (e.g. switch expression doesn't accept typeof as it is not constant).

```csharp
private static ModelType ExtractValue<ModelType>(Type modelType, JsonParseNode rootNode)
{
    object value;
    if (modelType == typeof(bool))
        value = rootNode.GetBoolValue();
    else if (modelType == typeof(string))
        value = rootNode.GetStringValue();
    else if (modelType == typeof(int))
        value = rootNode.GetIntValue();
    else if (modelType == typeof(float))
        value = rootNode.GetFloatValue();
    else if (modelType == typeof(double))
        value = rootNode.GetDoubleValue();
    else if (modelType == typeof(Guid))
        value = rootNode.GetGuidValue();
    else if (modelType == typeof(DateTimeOffset))
        value = rootNode.GetDateTimeOffsetValue();
    else
        throw new InvalidOperationException("error handling the response, unexpected type");

    return (ModelType)value;
}
```

Reply:
yeah the inability to use typeof for the switch made me sad too.
refactored to about duplicating the casting.